helix-term/src/commands.rs

```diff
@@ -1202,19 +1202,15 @@ fn goto_file_impl(cx: &mut Context, action: Action) {
     let (view, doc) = current_ref!(cx.editor);
     let text = doc.text();
     let selections = doc.selection(view.id);
+    let primary = selections.primary();
     let rel_path = doc
         .relative_path()
         .map(|path| path.parent().unwrap().to_path_buf())
         .unwrap_or_default();
-    let mut paths: Vec<_> = selections
-        .iter()
-        .map(|r| text.slice(r.from()..r.to()).to_string())
-        .collect();
-    let primary = selections.primary();
-    // Checks whether there is only one selection with a width of 1
-    if selections.len() == 1 && primary.len() == 1 {
-        paths.clear();
 
+    let paths: Vec<_> = if selections.len() == 1 && primary.len() == 1 {
+        // Secial case: if there is only one one-width selection, try to detect the
+        // path under the cursor.
```

Comment:
Might it be clearer if this block were pulled out into a function named something like `goto_file_detect_path`?